File: top-interview-questions-easy/String/reverse-integer/7.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        n = int(str(abs(x))[::-1])
        return 0 if x == 0 else ((x > 0) - (x < 0)) * n * (n < 2 ** 31)

File: algorithms/7. 整数反转/7.js
/**
 * @param {number} x
 * @return {number}
 */
const reverse = function(x) {
    const limit = 2 ** 31

    if (x > limit) return 0
    const bool = x > 0
    const array = String(Math.abs(x)).split('').reverse()
    const number = Number(array.join(''))

    if (number > limit) return 0
    return bool ? number : 0 - number
}
